add ProblemController and ProblemInterface

Lipek88 opened the pull request:
add .gitignore to GitHub

Lipek88 committed:
add ProblemController and ProblemInterface
add .gitignore to GitHub

DrJul commented on src/main/java/com/akademia/educators_back/controller/ProblemController.java:
```diff
@@ -0,0 +1,37 @@
+
+    public ProblemService problemService;
+
+    @GetMapping("/api/problemsList")
```
czyli nasza ścieżka to "api/problems/api/problemList" -> "api/problems/list" lub "api/problems"

DrJul commented on src/main/java/com/akademia/educators_back/controller/ProblemController.java:
```diff
@@ -0,0 +1,37 @@
+
+    @GetMapping("/api/problem/{id}")
+    public ResponseEntity<ProblemTo> findUserById(@PathVariable Long id){
+        try {
```
try catch służy do obsługi błędów, a tutaj robi za logikę na zasadzie if-else. Dobrą praktyką jest aby metoda miała jedno miejsce wyjścia.

DrJul commented on src/main/java/com/akademia/educators_back/controller/ProblemController.java:
```diff
@@ -0,0 +1,37 @@
+
+    @PostMapping("/api/addProblem")
+    void createProblem(@RequestBody ProblemTo problemTo){
+        problemService.addProblemToDB(problemTo);
```
Możnaby zwrócić status

DrJul commented on src/main/java/com/akademia/educators_back/service/ProblemInterface.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.util.List;
+
+public interface ProblemInterface {
```
Zamiast przyrostka Interface nie dawałbym nic, a przynajmniej nie to słowo. Jak mamy jakieś interfejsy w Javie to też nie nie mają tego przyrostka np: List

DrJul commented on src/main/java/com/akademia/educators_back/service/impl/ProblemService.java:
```diff
@@ -32,4 +38,20 @@ public void updateProblem(ProblemTo problemTo) {
+
+    @Override
+    public ProblemTo getProblemById(Long id) {
+        ProblemEntity problemEntity = problemRepository.findById(id).get();
```
a jeśli nic nie znajdziemy? wywoływanie get'a na optionalu jest nieprawidłowe.
1) najpierw dajemy isPresent() i wtedy get jeśli coś jest - ale to że jest uważane za antypattern dlatego odsyłam do nr2
2) .orElseThrow( ()-> exception) i to samo w metodzie wyżej gdzie robimy findAll

Lipek88 committed:
change mapping in ProblemController
fix try/catch in findUserById method
responseStatus in createProblemMethod
fix Interface name

Lipek88 committed:
fix try/catch in findUserById method

DrJul commented on src/main/java/com/akademia/educators_back/controller/ProblemController.java:
```diff
@@ -0,0 +1,41 @@
+        try {
+            responseEntity = ResponseEntity.ok().body(problemService.getProblemById(id));
+        }catch (ProblemDoesNotExistException e){
+            System.out.println("Problem does not exist");;
```
Mamy własny exception którego nie rzucamy tylko dajemy komunikat przez sout'a. Tak się nie robi, lepiej dać e.getMessage().

Lipek88 committed:
add DataLoader
change getProblemById method in ProblemController
add ProblemDoesNotExistAdvise
